packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart

```diff
@@ -431,6 +457,41 @@ class EnrollVerbHandler extends AbstractVerbHandler {
     }
   }
 
+  /// Checks whether an enrollment with the same appName and deviceName already exists for the given request.
+  /// If a matching enrollment is found, [AtEnrollmentException] exception is thrown.
+  /// Otherwise, the enrollment request is accepted.
+  @visibleForTesting
+  Future<void> validateEnrollmentRequest(EnrollParams enrollParams) async {
+    // Fetches all the enrollment keys from the keystore.
+    List<dynamic> enrollmentKeys = keyStore.getKeys(regex: 'enrollments');
+
+    // Iterate all the enrollment keys and verify if there exists an enrollment the same appName and deviceName combination
+    for (String key in enrollmentKeys) {
+      AtData atData = AtData();
+      try {
+        atData = await keyStore.get(key);
+      } on KeyNotFoundException {
+        logger.finest('An enrollment with $key does not exist or expired');
+      }
+      if (atData.data == null) {
+        continue;
+      }
+      EnrollDataStoreValue enrollDataStoreValue =
+          EnrollDataStoreValue.fromJson(jsonDecode(atData.data!));
+
+      if ((enrollParams.appName == enrollDataStoreValue.appName &&
+              enrollParams.deviceName == enrollDataStoreValue.deviceName) &&
+          (enrollDataStoreValue.approval?.state ==
+                  EnrollmentStatus.approved.name ||
+              enrollDataStoreValue.approval?.state ==
+                  EnrollmentStatus.pending.name)) {
+        String enrollmentId = key.substring(0, key.indexOf('.'));
+        throw AtEnrollmentException(
+            'Another enrollment with id $enrollmentId exists with the app name: ${enrollParams.appName} and device name: ${enrollParams.deviceName}');
```

Comment:
Please also include the state of the enrollment (pending or approved) in the message